### PR TITLE
105 UI

### PR DESCRIPTION
**Pull request by allishultes**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

See [#105 in Storybook](https://github.com/bbc/digital-paper-edit-storybook/issues/105) 

**Describe what the PR does**    

- [x] Adds warning if user tries to close DPE tab when media is uploading
- [x] ~~Adds prompt if user tries to navigate away from DPE while media is uploading~~ see https://github.com/bbc/digital-paper-edit-firebase/issues/284

**State whether the PR is ready for review or whether it needs extra work**    

Ready

**Additional context**    

Please review [with the related Storybook PR](https://github.com/bbc/digital-paper-edit-storybook/pull/106) to see all of the fixes for the relevant issue. 

We don't have options to customise the alerts / prompts, and so I have not been able to implement Ngan-Thi's designs exactly for these. We can only customise strings for the prompt message when users try to click links during uploads, and we only have custom string support on IE for the alert when the user tries to close the window. 

**To test:**
- Go to the Workspace view
- Upload a media file
- Try to close the tab when the file is uploading (an alert should appear)
- Try to click on:
  - A completed transcript link (an alert should appear)
  - A programme script link (an alert should appear)
  - The "back to projects" button (an alert should appear)
  - "Transcribe media" / "Create new programme script" (an alert should NOT appear)

<!-- 
## User Story / Context
|As a ...|I want ...|So that ...|
|-|-|-|
|<Who>|<What>|<Why>|

## Acceptance Criteria
- <Criteria to satisfy the PR Issue>

### CODE REVIEW

File: src/Components/Workspace/index.js [hunk 1]
```diff
@@ -91,6 +101,15 @@ const WorkspaceView = props => {
     return () => {};
   }, [ id, props.firebase ]);
 
+  useEffect(() => {
+    window.onbeforeunload = (event) => {
+      event.preventDefault();
+      if (uploadTasks.size !== 0) {
+        event.returnValue = 'Your file has not finished uploading';
```
Comment (allishultes) [inline src/Components/Workspace/index.js:108]: This only results in a custom message in Safari; there aren't ways to add custom messages for Firefox and Chome.

**Review by aniablaziak — APPROVED** [review]
Thanks Alli, this worked for me in Chrome, Firefox and Safari. 

Not related to this PR: I noticed that Safari is not letting me upload some file formats (.mp3, .mp4) 🤔

**Review by tamsingreen — CHANGES_REQUESTED** [review]
Code looks good - needs changes to only show the prompt on navigating away from DPE or closing the tab as upload continues when staying within the app :)

**Comment by allishultes** [comment]
> Code looks good - needs changes to only show the prompt on navigating away from DPE or closing the tab as upload continues when staying within the app :)

I'm actually not able to replicate this on all files, Tamsin! I've navigated within the app when uploading two test files yesterday, and neither have finished uploading or started transcribing ...

Screenshot 2021-06-10 at 09 15 17

Screenshot 2021-06-10 at 09 14 13

Screenshot 2021-06-10 at 09 13 38

On a separate note, ideally these should probably have `status:fail` since they will never finish...

**Review by tamsingreen — APPROVED** [review]
No prompt when I navigate elsewhere in the app; prompt when I try to close the tab 💫

**Pull request merged**